unity-renderer/Assets/DCLPlugins/ECS7/Systems/ScenesUiSystem/ECSScenesUiSystem.cs

```diff
@@ -88,14 +102,21 @@ public void Update()
                 currentScene ??= GetCurrentScene(currentSceneNumber, loadedScenes);
 
                 // we apply current scene UI
-                if (currentScene != null)
-                {
-                    if (ApplySceneUI(internalUiContainerComponent, uiDocument, currentScene, isSceneUIEnabled))
-                    {
-                        isPendingSceneUI = false;
-                    }
-                }
+                if (currentScene != null && ApplySceneUI(internalUiContainerComponent, uiDocument, currentScene, isSceneUIEnabled))
+                    isPendingSceneUI = false;
             }
+
+            componentWriter.PutComponent(
+                currentSceneNumber,
+                SpecialEntityId.SCENE_ROOT_ENTITY,
+                ComponentID.UI_CANVAS_INFORMATION,
+                new PBUiCanvasInformation()
+                {
+                    InteractableArea = interactableArea,
+                    Width = Screen.width,
+                    Height = Screen.height,
+                    DevicePixelRatio = CalculateDevicePixelRatio()
+                });
```

Comment:
do we need to write this component on every update?

Reply:
To avoid that we'd need to cache 2 `Vector2` to check on every frame if the viewport size changed or the real screen size changed, does that sound good?

Reply:
At a protocol level, it should be present for the tick#1 (getCrdtState/onStart/main) functions. Updates _can_ be sent on every frame but that's not strictly needed unless some value actually changes

Reply:
Yes, let's cache those vectors and compare them to check if values have changed, it would be far more cheaper than sending the component on every update, because probably, in most common scenario, values won't change during runtime

Reply:
Refactored separating the logic into a new UiCanvasInformation system, based on suggestions